### PR TITLE
Rgb/add onboarding documentation

### PR DESCRIPTION
**Pull request by adudenamedruby**
## Description (optional)

Adds iOS onboarding documentation to `experimenter-docs`

## Issue that this pull request resolves

[FXiOS ticket](https://github.com/mozilla-mobile/firefox-ios/issues/15059)

### CODE REVIEW

File: docs/onboarding/onboarding-feature-ios.md [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+
+Using Experimenter in the general case is documented elsewhere, so this document is specifically concerned with configuring the onboarding feature. [Example experiment](https://experimenter.services.mozilla.com/nimbus/lifestyles-images-onboarding-experiment/summary)
+
+# References
+For the most up-to-date configurations, the main code base will always be the best place to check.
+- [Nimbus manifest](https://github.com/mozilla-mobile/firefox-ios/blob/main/nimbus.fml.yaml)
+- [onboarding feature manifest](https://github.com/mozilla-mobile/firefox-ios/blob/main/nimbus-features/onboardingFrameworkFeature.yaml)
+
+⚠️ The onboarding feature is a **first run experiment**.
+- [for information on first run experiment](https://experimenter.info/mobile-first-run-experiments)
```
Comment (eliserichards) [inline docs/onboarding/onboarding-feature-ios.md:41]:
```suggestion
⚠️ The onboarding feature is a [first run experiment](https://experimenter.info/mobile-first-run-experiments)
```

File: docs/onboarding/onboarding-feature-ios.md [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+```
+
+# Feature Definition
+The onboarding feature is split into several values:
+- conditions
+- cards
+- dismissable
+
+## Conditions
+The conditions table is a set of triggers, like messaging, that can be used on cards to include or exclude them from appearing. The coditions list is a json of valid JEXL expressions. These expressions can be used in the `prerequisites` or the `disqualifiers` fields in cards. Eg:
```
Comment (eliserichards) [inline docs/onboarding/onboarding-feature-ios.md:114]:
```suggestion
The conditions table is a set of triggers, like messaging, that can be used on cards to include or exclude them from appearing. The conditions list is a json of valid JEXL expressions. These expressions can be used in the `prerequisites` or the `disqualifiers` fields in cards. E.g.:
```

File: docs/onboarding/onboarding-feature-ios.md [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+|:---------------------------|:------------------------------------------------------------------------------------|
+| next-card                  | Will take the user to the next card                                                 |
+| sync-sign-in               | Will take the user to the sync sign in flow                                         |
+| request-notifications      | Will request to allow notifications from the user                                   |
+| set-default-browser        | Will send the user to settings to set Firefox as their default browser              |
+| open-default-browser-popup | Will open up a popup with instructions for setting Firefox as their default browser |
+| read-privacy-policy        | Will open a webview where the user can read the privacy policy                      |
+
+## Dismissable
+This is a property for the whole onboarding, and controls whether there is an `x` at the top of the screen or not. This is a simple boolean value. ex:
```
Comment (eliserichards) [inline docs/onboarding/onboarding-feature-ios.md:169]:
```suggestion
This is a property for the whole onboarding, and controls whether there is an `x` at the top of the screen or not. This is a simple boolean value:
```

File: docs/onboarding/onboarding-feature-ios.md [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+
+- the onboarding feature may be under active development
+- card types may be added
+- card attributes may be added
+
+# Scene setting
+
+The onboarding feature is built on top of Nimbus, Mozilla's experimentation platform. Nimbus allows you to send bits of configuration to application features from Experimenter, the web-application staff use to launch and manage experiments and rollouts.
+
+Using Experimenter in the general case is documented elsewhere, so this document is specifically concerned with configuring the onboarding feature. [Example experiment](https://experimenter.services.mozilla.com/nimbus/lifestyles-images-onboarding-experiment/summary)
```
Comment (eliserichards) [inline docs/onboarding/onboarding-feature-ios.md:33]:
```suggestion
Using Experimenter in the general case is documented elsewhere, so this document is specifically concerned with configuring the onboarding feature. Example experiment: [Lifestyles images onboarding experiment](https://experimenter.services.mozilla.com/nimbus/lifestyles-images-onboarding-experiment/summary)
```

File: docs/onboarding/onboarding-feature-ios.md [hunk 5]
```diff
@@ -0,0 +1,249 @@
...
+
+# References
+For the most up-to-date configurations, the main code base will always be the best place to check.
+- [Nimbus manifest](https://github.com/mozilla-mobile/firefox-ios/blob/main/nimbus.fml.yaml)
+- [onboarding feature manifest](https://github.com/mozilla-mobile/firefox-ios/blob/main/nimbus-features/onboardingFrameworkFeature.yaml)
```
Comment (eliserichards) [inline docs/onboarding/onboarding-feature-ios.md:38]:
```suggestion
- [Onboarding feature manifest](https://github.com/mozilla-mobile/firefox-ios/blob/main/nimbus-features/onboardingFrameworkFeature.yaml)
```

File: docs/onboarding/onboarding-feature-ios.md [hunk 6]
```diff
@@ -0,0 +1,249 @@
...
+| type          | fresh-install                                         |
+| title         | Onboarding/Onboarding.Welcome.Title.v114              |
+| body          | Onboarding/Onboarding.Welcome.Description.v114        |
+| link          | title: Onboarding/Onboarding.Welcome.Link.Action.v114 |
+|               | url: "https://www.mozilla.org/privacy/firefox/"       |
+| image         | welcome-globe                                         |
+| ordering      | 10                                                    |
+| buttons       |                                                       |
+| - primary     | title: Onboarding/Onboarding.Welcome.Action.v114      |
+|               | action: next-card                                     |
```
Comment (eliserichards) [inline docs/onboarding/onboarding-feature-ios.md:218]: Should `url:` and `action` be in the first column here? 
image

Comment (eliserichards) [reply docs/onboarding/onboarding-feature-ios.md:218]: Same question for `Default Sync card` and `Default Notification card`

**Review by eliserichards — COMMENTED** [review]
This is awesome! Thank you so much for writing this up! I have a few questions and a few nits (feel free to ignore those if you want):

File: docs/onboarding/onboarding-feature-ios.md (hunk 6 above)
Comment (adudenamedruby) [reply docs/onboarding/onboarding-feature-ios.md:218]: Dang it! I mixed up two things. I'll fix it!

**Review by eliserichards — APPROVED** [review]
Looks great 🚀

**Pull request merged**